src/codegen/mod.rs

```diff
@@ -2172,6 +2177,206 @@ impl CodeGenerator for CompInfo {
+                    }
+                },
+                _ => (elem),
+            }
+        }
+    ).collect()
+}
+
+impl Method {
+    /// If self is an overloaded operator that this function recognizes, this
+    /// function will put the correct wrapper into result and return true. Note
+    /// that this function does not need to recognize all overloaded operators
+    /// for Bindgen to work correctly, because this function "only" adds some
+    /// syntactic sugar to make the Rust code that calls the C++ code using
+    /// Bindgen look a bit nicer. A non-exhaustive list of ignored operators:
+    /// * `operator[]` - We could translate this C++ operator to Index or
+    ///   IndexMut, but in Rust usually the [] operator does the bounds
+    ///   checking, but in C++ usually the caller does the bounds checking. This
+    ///   might lead to bugs if the Rust caller thinks that the C++ operator
+    ///   will do bounds checking.
+    /// * `operator<, operator>, operator<=, operator>=` Rust does have the
+    ///   "Ord" and "PartialOrd" Traits that are similar to those operators, but
+    ///   those operators would require more coding work from the bindgen
+    ///   Authors than other operators and I'm lazy.
+    /// * `operator!=` In C++ you could (but you probably should not) define a
+    ///   != Operator and a == Operator, so that a != b is not !(a == b). You
+    ///   cannot do that in Rust, because Rustc translates a != b to !(a == b).
+    ///   We ignore the != Operator that the C++ code defined and just use the
+    ///   == Operator that the C++ code defined.
+    fn try_codegen_operator (
+        &self,
+        ctx: &BindgenContext,
+        ty_for_impl: &proc_macro2::TokenStream,
+        result: &mut CodegenResult,
+    ) {
+        // We start by getting the name of the function.
+        let function_item = ctx.resolve_item(self.signature());
+        let function = function_item.expect_function();
+        let name = function.name().to_owned();
+        if !name.starts_with("operator") {
+            return;
+        }
+
+        // We then get the type of the return value (ret_type) and the type of
+        // the second argument (second_arg_type) (if a second argument exists).
+        let function_name = ctx.rust_ident(function_item.canonical_name(ctx));
+        let signature_item = ctx.resolve_item(function.signature());
+        let signature = match *signature_item.expect_type().kind() {
+            TypeKind::Function(ref sig) => sig,
+            _ => panic!("How in the world?"),
+        };
+        let return_item = ctx.resolve_item(signature.return_type());
+        let ret_type = return_item.to_rust_ty_or_opaque(ctx, &());
+        let args = signature.argument_types();
+        let second_arg_type = if args.len() < 2 {
+            None
+        } else {
+            dbg!("test");
+            let temp = utils::type_id_to_rust_type(ctx, signature.argument_types()[1].1);
```

Comment:
Can't we dig here if there is a pointer instead here?

Reply:
What do you mean with "digging"?